dali/kernels/slice/slice_cpu.h

```diff
@@ -287,10 +382,27 @@ class SliceCPU {
     return req;
   }
 
+  /**
+   * @brief Run the kernel with an execution engine.
+   *
+   *        For execution engines other than
+   *        SequentialExecutionEngine, the algorithm will try to split the slice into
+   *        similar sized blocks until we either reach a minimum of ``req_nblocks`` or
+   *        the block volume is smaller than the minimum practical size, ``min_blk_sz``.
+   * @param context Kernel context
+   * @param out Output tensor view
+   * @param in Input tensor view
+   * @param slice_args Slice arguments
+   * @param min_blk_sz Minimum practical block size
+   * @param req_nblocks Requested number of blocks
+   */
+  template <typename ExecutionEngine = SequentialExecutionEngine>
   void Run(KernelContext &context,
            OutTensorCPU<OutputType, Dims> &out,
            const InTensorCPU<InputType, Dims> &in,
-           const SliceArgs<OutputType, Dims> &slice_args) {
+           const SliceArgs<OutputType, Dims> &slice_args,
+           ExecutionEngine &exec_engine,
+           int min_blk_sz = 16000, int req_nblocks = 32) {
```

Comment:
```suggestion
           int min_blk_sz = 16000, int req_nblocks = -1) {
```
I think you can calculate the minimum number of blocks internally, from `engine.NumThreads()`.